cpp/src/arrow/compute/exec/expression.h

```diff
@@ -131,15 +132,15 @@ inline bool operator!=(const Expression& l, const Expression& r) { return !l.Equ
 // Factories
 
 ARROW_EXPORT
-Expression literal(Datum lit);
+Expression literal(Datum&& lit);
```

Comment:
Please revert this; it's not illegal for these factories to accept a copy:
```c++
const Datum& copyable = GetDatumRefFromSomewhere();
Expression expr = literal(copyable);  // will fail to compile if literal() only accepts Datum&&
```

Reply:
Oops, sorry. I think I went overboard with these changes.